build-logic/gradle/src/main/kotlin/com/avito/android/CheckCommonProperties.kt

```diff
@@ -0,0 +1,114 @@
+package com.avito.android
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectories
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+abstract class CheckCommonProperties : DefaultTask() {
+
+    @get:InputFile
+    abstract val commonPropertiesFile: RegularFileProperty
+
+    @get:OutputDirectories
+    abstract val projectDirs: SetProperty<Directory>
+
+    @TaskAction
+    fun doWork() {
+        val commonProperties = PropertiesConfiguration(commonPropertiesFile.get().asFile)
+
+        val inconsistencies = mutableListOf<Inconsistency>()
+
+        projectDirs.get().forEach { projectDir ->
+
+            val projectPath = relativePath(projectDir.asFile)
+
+            val gradlePropertiesFile = File(projectDir.asFile, "gradle.properties")
+
+            val properties = if (gradlePropertiesFile.exists()) {
+                PropertiesConfiguration(gradlePropertiesFile)
+            } else {
+                inconsistencies.add(
+                    Inconsistency(
+                        projectPath = projectPath,
+                        reason = "gradle.properties for project $projectDir not found"
+                    )
+                )
+                return@forEach
+            }
+
+            commonProperties.keys.forEach { key ->
+                if (!properties.containsKey(key)) {
+                    inconsistencies.add(
+                        Inconsistency(
+                            projectPath = projectPath,
+                            reason = "missing property: $key"
+                        )
+                    )
+                } else {
+                    val commonValue = commonProperties.getProperty(key)
+                    val value = properties.getProperty(key)
+                    if (value != commonValue) {
+                        inconsistencies.add(
+                            Inconsistency(
+                                projectPath = projectPath,
+                                reason = "different value for property: $key\n" +
+                                    "     expected: $commonValue\n" +
+                                    "     actual  : $value"
+                            )
```

Comment:
Does it mean that we can't override common properties?
Didn't find use cases

Reply:
Yes, we can't
It shouldn't be a common property in that case